spec/lib/validator_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require "govuk_schemas/validator"
+
+RSpec.describe GovukSchemas::Validator do
+  describe "#valid?" do
+    it "detects an valid schema" do
+      example = GovukSchemas::RandomExample.for_schema(publisher_schema: "placeholder")
+      validator = described_class.new("placeholder", "publisher", example)
+
+      expect(validator.valid?).to eq true
+    end
+
+    it "detects an invalid schema" do
+      example = { obviously_invalid: true }
+      validator = described_class.new("placeholder", "publisher", example)
+
+      expect(validator.valid?).to eq false
+    end
+  end
+
+  describe "#error_message" do
+    let(:start_of_error_message) do
+      <<~DOC
+        expected the payload to be valid against the 'placeholder' schema:
+
+        {
+          "obviously_invalid": true
+        }
+
+        Validation errors:
+      DOC
+    end
+
+    it "constructs an error message based on the schema_name, payload and errors" do
+      example = { obviously_invalid: true }
+      validator = described_class.new("placeholder", "publisher", example)
+      expect(validator.error_message).to include start_of_error_message
+      expect(validator.error_message).to match(/- The item did not contain a required property of '([a-z_]+)' in schema placeholder/i)
+      expect(validator.error_message).to match(/- The item contains additional properties ([a-z_" \[\]]+) outside of the schema when none are allowed in schema placeholder/i)
```

Comment:
is the regex  capture`([a-z_" \[\]]+)` matching `obviously_invalid`? I think it'd be a bit clearer to call that out directly as I found myself trying to work out what the regex meant but I think it's just matching our input?

Reply:
Oh sure. Shall we just have one invalid field here to make it simpler? After all making it more complex is only testing the json-schema gem.

Reply:
It doesn't look like this is fully changed? I can still see the complex condition and this remains: https://github.com/alphagov/govuk_schemas/pull/66/files#diff-2f7804136ad659dabee755e224bb9f76e03f0b8f798438651d4a57d3ae732b57R35

Reply:
Sorry i've pushed up now @kevindew